langchain/vectorstores/mongodb_atlas.py

```diff
@@ -0,0 +1,242 @@
+            pre_filter: Optional Dictionary of argument(s) to prefilter on document
+                fields.
+            post_filter_pipeline: Optional Pipeline of MongoDB aggregation stages
+                following the knnBeta search.
+
+        Returns:
+            List of Documents most similar to the query and score for each
+        """
+        pipeline = [
+            {
+                "$search": {
+                    "index": self._index_name,
+                    "knnBeta": {
+                        "vector": self._embedding.embed_query(query),
+                        "path": self._embedding_key,
+                        "k": k,
+                        "filter": pre_filter or {},
+                    },
+                }
+            },
+            {"$project": {"score": {"$meta": "searchScore"}, self._embedding_key: 0}},
+        ]
+        if post_filter_pipeline is not None:
+            pipeline.extend(post_filter_pipeline)
+        cursor = self._collection.aggregate(pipeline)
+        docs = []
+        for res in cursor:
+            text = res.pop(self._text_key)
+            score = res.pop("score")
+            docs.append((Document(page_content=text, metadata=res), score))
+        return docs
+
+    def similarity_search(
+        self,
+        query: str,
+        k: int = 4,
+        pre_filter: Optional[dict] = None,
+        post_filter_pipeline: Optional[List[Dict]] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Return MongoDB documents most similar to query.
+
+        Use the knnBeta Operator available in MongoDB Atlas Search
+        This feature is in early access and available only for evaluation purposes, to
+        validate functionality, and to gather feedback from a small closed group of
+        early access users. It is not recommended for production deployments as we may
+        introduce breaking changes.
+        For more: https://www.mongodb.com/docs/atlas/atlas-search/knn-beta
+
+        Args:
+            query: Text to look up documents similar to.
+            k: Optional Number of Documents to return. Defaults to 4.
+            pre_filter: Optional Dictionary of argument(s) to prefilter on document
+                fields.
+            post_filter_pipeline: Optional Pipeline of MongoDB aggregation stages
+                following the knnBeta search.
+
+        Returns:
+            List of Documents most similar to the query and score for each
+        """
+        docs_and_scores = self.similarity_search_with_score(
+            query,
+            k=k,
+            pre_filter=pre_filter,
+            post_filter_pipeline=post_filter_pipeline,
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    @classmethod
+    def from_texts(
+        cls,
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        connection_string: Optional[str] = None,
```

Comment:
This should not be Optional. It's a mandatory parameter:
connection_string: str,

Reply:
Also note that initiating a new MongoClient (the user may very likely already have one defined for their entire application) is costly since it involves TLS handshakes, authentications and a setup of a new connection pool (impact the memory on both end - application and database server). This is the primary reason why I used the client: MongoClient parameter instead so it could be passed in.

Reply:
Better to pass in the mongo client and to use the __init__ method instead to get back the vectorstore

Reply:
ah thanks for info, let's switch to client then! should keep these as kwargs though because this method is inherited

Reply:
Perfect